compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java

```diff
@@ -181,9 +181,10 @@ void instrumentPackage(BIRPackage pkg) {
      * @param pkg The package containing the function
      */
     private void rewriteAsyncInvocations(BIRFunction func, BIRTypeDefinition attachedTypeDef, BIRPackage pkg) {
-        Name org = new Name(IdentifierUtils.decodeIdentifier(pkg.org.value));
-        Name module = new Name(IdentifierUtils.decodeIdentifier(pkg.name.value));
-        PackageID currentPkgId = new PackageID(org, module, pkg.version);
+        PackageID packageID = pkg.packageID;
+        Name org = new Name(IdentifierUtils.decodeIdentifier(packageID.orgName.getValue()));
+        Name module = new Name(IdentifierUtils.decodeIdentifier(packageID.name.getValue()));
+        PackageID currentPkgId = new PackageID(org, module, packageID.version);
```

Comment:
can we use `pkg.packageID` here?

Reply:
This is created after decoding the org and name values. So I guess `pkg.packageID` and `currentPkgId` could be different